src/mock.rs

```diff
@@ -214,7 +214,24 @@ pub struct Mock {
     // If `Some(max_n_matches)`, when `max_n_matches` matching incoming requests have been processed,
     // `self.matches` should start returning `false`, regardless of the incoming request.
     pub(crate) max_n_matches: Option<u64>,
-    pub(crate) expectation: Times,
+    pub(crate) expectation: Expectation,
+}
+
+#[derive(Clone)]
+pub(crate) struct Expectation {
+    /// expected range
+    pub(crate) range: Times,
+    /// message displayed when `range` expectation is not met.
+    pub(crate) error_message: String,
+}
+
+impl Default for Expectation {
+    fn default() -> Self {
+        Self {
+            range: Times(TimesEnum::Unbounded(RangeFull)),
+            error_message: "".to_string(),
```

Comment:
```suggestion
            // No need to specify an error message because the number of requests will
            // always fall within `range` given that the range is unbounded!
            error_message: "".to_string(),
```